src/electionguard/elgamal.cpp

```diff
@@ -258,4 +264,388 @@ namespace electionguard
         return make_unique<ElGamalCiphertext>(move(resultPad), move(resultData));
     }
 
+
+#pragma region HashedElGamalCiphertext
+    
+    struct HashedElGamalCiphertext::Impl {
+        unique_ptr<ElementModP> pad;
+        vector<uint8_t> ciphertext;
+        vector<uint8_t> mac;
+
+        Impl(unique_ptr<ElementModP> pad, vector<uint8_t> ciphertext, vector<uint8_t> mac)
+            : pad(move(pad)), ciphertext(ciphertext), mac(mac)
+        {
+        }
+
+        [[nodiscard]] unique_ptr<HashedElGamalCiphertext::Impl> clone() const
+        {
+            auto _pad = make_unique<ElementModP>(*pad);
+            return make_unique<HashedElGamalCiphertext::Impl>(move(_pad), ciphertext, mac);
```

Comment:
Can we just use Clone here instead of making a new smart pointer and then moving the data?

Reply:
Sure, we can do that. I was following the convention used in the ElGamalCiphertext::Impl above, but either way works for me.

Reply:
I am reviewing and looking into what was done before to make sure that we are sure of the patterns used.